src/storage/v3/shard.cpp

```diff
@@ -1075,9 +1077,14 @@ bool Shard::CreateIndex(LabelId label, const std::optional<uint64_t> desired_com
 }
 
 bool Shard::CreateIndex(LabelId label, PropertyId property, const std::optional<uint64_t> desired_commit_timestamp) {
-  // TODO Fix Index
-  // if (!indices_.label_property_index.CreateIndex(label, property, labelspace.access())) return false;
-  return false;
+  if (label == primary_label_ && schemas_.GetSchema(primary_label_)->second.size() == 1 &&
+      schemas_.GetSchema(primary_label_)->second[0].property_id == property) {
+    // Index already exists on primary key
```

Comment:
It is a good question whether we say it exists in this case or the user cannot create such an index. I am not sure which is better. For M1 I guess this is fine, but later on we have to make this clear. Let's put a TODO here I think.